src/main/java/org/jenkinsci/test/acceptance/plugins/credentials/CredentialsRESTClient.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2023 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.test.acceptance.plugins.credentials;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.LaxRedirectStrategy;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * Client to interact with Credentials REST API.
+ */
+public class CredentialsRESTClient {
+
+    private static final String BASE_REST_URL = "credentials/store/system";
+    private static final String BASE_DOMAIN_URL = BASE_REST_URL + "/domain";
+
+    private static final String BASE_CREDENTIALS_URL = BASE_DOMAIN_URL + "/_/credential/%s/config.xml";
+    private static final String CREATE_CREDENTIALS_URL = BASE_DOMAIN_URL + "/_/createCredentials";
+    private static final String BASE_DOMAINS_URL = BASE_DOMAIN_URL + "/%s/config.xml";
+    private static final String CREATE_DOMAINS_URL = BASE_REST_URL + "/createDomain";
+
+    private static final String CONTENT_TYPE_HEADER_NAME = "Content-Type";
+    private static final String CONTENT_TYPE = "application/xml;charset=utf-8";
+
+    private final HttpClient httpClient;
+    private final URL jenkinsUrl;
+
+
+    public CredentialsRESTClient(final URL jenkinsUrl) {
```

Comment:
I am wondering why we should be doing this via the REST API and not via the Jenkins UI?
if we want to test the RestAPI we can do that inside the credentials plugin itself in a `RealJenkinsRule`

Reply:
Yea, it is weird that it is not using the UI to create the credentials, the only explanation I can come up with for now  is speed.